test/boost/sstable_test.cc

```diff
@@ -63,26 +63,17 @@ SEASTAR_TEST_CASE(composite_index) {
     return test_using_working_sst(composite_schema(), "test/resource/sstables/composite");
 }
 
-template<typename Func>
-inline future<>
-test_using_reusable_sst(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
-    return test_env::do_with_async([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
-        auto sst = env.reusable_sst(std::move(s), std::move(dir), generation_from_value(gen)).get();
-        func(env, std::move(sst));
-    });
-}
-
-template<typename T, typename Func>
-inline future<T>
-test_using_reusable_sst_returning(schema_ptr s, sstring dir, sstables::generation_type::int_t gen, Func&& func) {
-    return test_env::do_with_async_returning<T>([s = std::move(s), dir = std::move(dir), gen, func = std::move(func)] (test_env& env) {
+inline auto
```

Comment:
I don't like returning auto. It means the return type can only be known by reading the function body or documentation. Here it can't be guessed from the name.

Please use something like invoke_result_t. It means abandoning the nice abbreviated template syntax, but it restores readability.